### DISCUSSION TITLE
How to do something on bot shutdown (got term signal) with 3.x version?

### DISCUSSION THREAD

**Discussion by Hudrolax** [aiogram/aiogram#1201] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by NiKuma0** [comment, marked as answer]
```python

dp = Dispatcher(...)

@dp.shutdown()
async def on_shutdown(*args, **kwargs):
  ...

@dp.startup()
async def on_startup(*args, **kwargs):
  ...
```
I recommend asking this type of question in the Aiogram Telegram chat [@aiogram](https://t.me/aiogram)
